op-service/dial: Add WaitRollupSync

sebastianst opened the pull request:
**Description**

`WaitRollupSync` is a standalone function that can be used to wait for the given rollup node to have synced to the provided L1 block.

**Tests**

Added tests of all possible cases.

**Additional context**

This is going to be used by the batcher and proposer in a follow up PR to wait at startup for the sequencer to sync to tip.

**Metadata**

- Towards https://github.com/ethereum-optimism/client-pod/issues/698

trianglesphere commented on op-service/dial/rollupclient_interface.go:
```diff
@@ -11,10 +11,16 @@ import (
+
+// RollupSyncer is the interface of a rollup client from which its sync status
+// can be queried.
+type RollupSyncer interface {
```
`RollupSyncer` is a slightly weird name. Maybe `RollupSyncStatus` or `SyncStatusProvider` - I get what you're for with `Syncer`

sebastianst replied:
True, picked `SyncStatusProvider`

trianglesphere commented on op-service/dial/rollup_sync.go:
```diff
@@ -0,0 +1,37 @@
+	"github.com/ethereum/go-ethereum/log"
+)
+
+func WaitRollupSync(ctx context.Context, lgr log.Logger,
```
nit: Splitting the function args like this is a bit weird. It's a little long for a single line or it could be split with one arg per line.

IDK if the setup is worth it, but `lgr, rollup, & pollDuration` could all be init params on something that had a `Wait(ctx,target)` method

axelKingsley replied:
+1 for "one arg per line" when the signature is too long.

sebastianst replied:
Split the args up into individual lines.

Yeah there could be a `SyncStatusWaiter` with a `Wait` function, but since all use cases of this that I envision only ever call this once, I'd rather just have a single function. I suggest for this iteration we leave it with the function, and if we ever see a case for reuse, or splitting up the initialization from the waiting, we create that `SyncStatusWaiter`.

trianglesphere reviewed (commented):
Testing + logic LGTM. Just some comments on names

sebastianst committed:
op-service/dial: Add WaitRollupSync

This is going to be used by the batcher and proposer in a follow up PR to wait at startup for the sequencer to sync to tip.
